calvin/calvin.py

```diff
@@ -1,151 +1,146 @@
-  parser.add_argument("-m", "--max_batch", type=int, default=25)
-  args = parser.parse_args()
-
-  n = args.number
-  batches = []
-  while n > 0:
-      take = min(n, args.max_batch)
-      batches.append(take)
-      n -= take
-
-  choices = []
-  for b in tqdm.tqdm(batches):
-      completion = openai.Completion.create(
-          prompt=prompt,
-          n=b,
-          engine=args.engine,
-          max_tokens=300,
-          stop=stop,
-          temperature=data[prompt][0]['temperature'],
-          logprobs=0,
-      )
-      choices += completion.choices
-
-  def score(a):
-      return sum(a.logprobs.token_logprobs) / len(a.logprobs.token_logprobs)
-
-  choices = sorted(choices, key=lambda a: -score(a))
-  for i, choice in enumerate(choices):
-      print(f"======")
-      print(choice.text.strip())
-
-  return 0
-
-if __name__ == "__main__":
-    prompt = input('Enter prompt: ')
-    sys.exit(main(prompt))
\ No newline at end of file
+# Defile module-scoped variables
+engine = "davinci"
+headers = {}
+prompts = {}
+
+
+# Initialize module
+def initialize(_api_key, _engine):
+  global engine, headers, prompts
+
+  if _engine:
+    engine = _engine
+
+  headers = {
+    "Content-Type": "application/json",
+    "Authorization": f"Bearer {_api_key}"
+  }
+
+  prompts = json.load(open("./calvin/prompts.json"))
+
+#
+def get_globals():
+  global engine, headers, prompts
+
+  print({
+    engine: engine,
+    headers: headers,
+    prompts: prompts # Figure out why this wont serialize -_-
+  })
+
+# Change engine setting
+def set_engine(_engine):
+  global engine
+  engine = _engine
+
+# List Engines GET
+# Lists the currently available engines, and provides basic information about each option such as the owner and availability.
+def list_engines():
+  global headers
+
+  url = "https://api.openai.com/v1/engines"
+  response = requests.get(url, headers=headers)
+
+  if response.ok:
+    print(f"JSON: {response.json()}")
+    return response.json()
+  else:
+    return response.raise_for_status()
+
+# Retrieve Engine GET
+# Retrieves an engine instance, providing basic information about the engine such as the owner and availability.
+def retrieve_engine():
+  global headers
+
+  url = f"https://api.openai.com/v1/engines/{engine}"
+  response = requests.get(url, headers=headers)
+
+  if response.ok:
+    print(f"JSON: {response.json()}")
+    return response.json()
+  else:
+    return response.raise_for_status()
+
+# Create Completion POST
+# Returns new text as well as, if requested, the probabilities over each alternative token at each position.
+def create_completion(payload):
+  global engine, headers
+
+  url = f"https://api.openai.com/v1/engines/{engine}/completions"
+  response = requests.post(url, headers=headers, json=payload)
+
+  if response.ok:
+    print(f"JSON: {response.json()}")
+    return response.json()
+  else:
+    return response.raise_for_status()
+
+def complete_prompt(prompt):
+  payload = {
```

Comment:
How would the end-user go about adjusting the hyperparameter settings here? They would need to call `create_completion` and pass it a json, no?

If that's the case, we should set these defaults as something more user friendly. `max_tokens = 5` won't produce very useful prompts out of the box. 

I'd suggest:
max_tokens = 150
temperature = 0.9
top_p = 1
n = 1